deployment/src/main/java/io/quarkiverse/jackson/jq/deployment/JacksonJqFunctionLoader.java

```diff
@@ -0,0 +1,152 @@
+            throws IOException {
+        return loadConfig(loader, CONFIG_PATH)
+                .stream()
+                .flatMap(j -> j.functions.stream())
+                .filter(j -> j.version != null && j.version.contains(version))
+                .collect(Collectors.toList());
+    }
+
+    public static List<JqJson> loadConfig(final ClassLoader loader, final String path) throws IOException {
+        final List<JqJson> result = new ArrayList<>();
+        final Enumeration<URL> iter = loader.getResources(path);
+        while (iter.hasMoreElements()) {
+            final StringBuilder buffer = new StringBuilder();
+            try (final BufferedReader reader = new BufferedReader(
+                    new InputStreamReader(iter.nextElement().openStream(), StandardCharsets.UTF_8))) {
+                while (true) {
+                    final String line = reader.readLine();
+                    if (line == null)
+                        break;
+                    if (line.startsWith("#"))
+                        continue;
+                    buffer.append(line);
+                    buffer.append('\n');
+                }
+            }
+            final MappingIterator<JqJson> iter2 = DEFAULT_MAPPER
+                    .readValues(DEFAULT_MAPPER.getFactory().createParser(buffer.toString()), JqJson.class);
+            while (iter2.hasNext()) {
+                result.add(iter2.next());
+            }
+        }
+        return result;
+    }
+
+    private static String[] extractFunctionNamesFromAnnotationIfVersionMatch(Function fn, final Version version) {
+        final BuiltinFunction annotation = fn.getClass().getAnnotation(BuiltinFunction.class);
+        if (annotation == null)
+            return null;
+        if (!annotation.version().isEmpty()) {
+            final VersionRange range = VersionRange.valueOf(annotation.version());
+            if (!range.contains(version))
+                return new String[0];
+        }
+        return annotation.value();
+    }
+
+    @SuppressWarnings("deprecation")
+    private static String[] extractFunctionNamesFromDeprecatedAnnotationIfVersionMatch(Function fn, final Version version) {
+        final net.thisptr.jackson.jq.internal.BuiltinFunction annotation = fn.getClass()
+                .getAnnotation(net.thisptr.jackson.jq.internal.BuiltinFunction.class);
+        if (annotation == null)
+            return null;
+        if (!annotation.version().isEmpty()) {
+            final VersionRange range = VersionRange.valueOf(annotation.version());
+            if (!range.contains(version))
+                return new String[0];
+        }
+        return annotation.value();
+    }
+
+    /**
+     * Do not use this method. This method is only for Quarkus extension.
+     */
+    public static Map<String, Function> loadFunctionsFromServiceLoader(final ClassLoader classLoader, final Version version) {
```

Comment:
We are the reason for this method, can't we use that directly? This way we copy less code from their lib.

Reply:
This is just a temporary thing. I've copied it to be user to be able to change it if needed but need to be refactored depending on the direction we sill take